### ISSUE TITLE
Bazel build failure with gpu version

### ISSUE THREAD

**Issue by uglybachelor** [tensorflow/tensorflow#11884]
------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: 
No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**:
source
- **TensorFlow version (use command below)**:
1.2.0, I'm installing from tha master branch
- **Python version**: 
2.7
- **Bazel version (if compiling from source)**:
0.5.3
- **CUDA/cuDNN version**:
5.1
- **GPU model and memory**:
GTX1060 6G
- **Exact command to reproduce**:
`bazel build --config=opt --config=cuda //tensorflow/tools/pip_package:build_pip_package`

### Describe the problem
I followed every step from the installing & configuration page at tf official [website](https://www.tensorflow.org/install/install_sources#ConfigureInstallation). But when I came the bazel build step, I ran into the problem showed in the error logs below.

I should mention that:

1. I choosed cuda support in the ` ./configure ` step, you can find the configure details below as well.
2. I previously installed the cpu version of tensorflow on the same computer and it worked fine. Now I've uninstalled it of course.

### Source code / logs

My configuration:
```
Please specify the location of python. [Default is /usr/bin/python]: 
Found possible Python library paths:
/usr/local/lib/python2.7/dist-packages
/usr/lib/python2.7/dist-packages
Please input the desired Python library path to use.  Default is /usr/local/lib/python2.7/dist-packages
Do you wish to build TensorFlow with jemalloc as malloc support? [Y/n]: 
jemalloc as malloc support will be enabled for TensorFlow.

Do you wish to build TensorFlow with Google Cloud Platform support? [y/N]: 
No Google Cloud Platform support will be enabled for TensorFlow.

Do you wish to build TensorFlow with Hadoop File System support? [y/N]: 
No Hadoop File System support will be enabled for TensorFlow.

Do you wish to build TensorFlow with XLA JIT support? [y/N]: 
No XLA JIT support will be enabled for TensorFlow.

Do you wish to build TensorFlow with VERBS support? [y/N]: 
No VERBS support will be enabled for TensorFlow.

Do you wish to build TensorFlow with OpenCL support? [y/N]: 
No OpenCL support will be enabled for TensorFlow.

Do you wish to build TensorFlow with CUDA support? [y/N]: y
CUDA support will be enabled for TensorFlow.

Please specify the CUDA SDK version you want to use, e.g. 7.0. [Leave empty to default to CUDA 8.0]: 
Please specify the location where CUDA 8.0 toolkit is installed. Refer to README.md for more details. [Default is /usr/local/cuda]: 
"Please specify the cuDNN version you want to use. [Leave empty to default to cuDNN 6.0]: 5
Please specify the location where cuDNN 5 library is installed. Refer to README.md for more details. [Default is /usr/local/cuda]:
Please specify a list of comma-separated Cuda compute capabilities you want to build with.
You can find the compute capability of your device at: https://developer.nvidia.com/cuda-gpus.
Please note that each additional compute capability significantly increases your build time and binary size. [Default is: 6.1]
Do you want to use clang as CUDA compiler? [y/N]: 
nvcc will be used as CUDA compiler.

Please specify which gcc should be used by nvcc as the host compiler. [Default is /usr/bin/gcc]: 
Do you wish to build TensorFlow with MPI support? [y/N]: 
No MPI support will be enabled for TensorFlow.

Please specify optimization flags to use during compilation when bazel option "--config=opt" is specified [Default is -march=native]: 
Add "--config=mkl" to your bazel command to build with MKL support.
Please note that MKL on MacOS or windows is still not supported.
If you would like to use a local MKL instead of downloading, please set the environment variable "TF_MKL_ROOT" every time before build.
Configuration finished

```
My command which introduced the error below:
`bazel build --config=opt --config=cuda //tensorflow/tools/pip_package:build_pip_package`

My error message:

```
.......
ERROR: Skipping '//tensorflow/tools/pip_package:build_pip_package': error loading package 'tensorflow/tools/pip_package': Encountered error while reading extension file 'cuda/build_defs.bzl': no such package '@local_config_cuda//cuda': Traceback (most recent call last):
	File "/home/zzh/tensorflow/third_party/gpus/cuda_configure.bzl", line 1039
		_create_local_cuda_repository(repository_ctx)
	File "/home/zzh/tensorflow/third_party/gpus/cuda_configure.bzl", line 976, in _create_local_cuda_repository
		_host_compiler_includes(repository_ctx, cc)
	File "/home/zzh/tensorflow/third_party/gpus/cuda_configure.bzl", line 145, in _host_compiler_includes
		get_cxx_inc_directories(repository_ctx, cc)
	File "/home/zzh/tensorflow/third_party/gpus/cuda_configure.bzl", line 120, in get_cxx_inc_directories
		set(includes_cpp)
depsets cannot contain mutable items
WARNING: Target pattern parsing failed.
ERROR: error loading package 'tensorflow/tools/pip_package': Encountered error while reading extension file 'cuda/build_defs.bzl': no such package '@local_config_cuda//cuda': Traceback (most recent call last):
	File "/home/zzh/tensorflow/third_party/gpus/cuda_configure.bzl", line 1039
		_create_local_cuda_repository(repository_ctx)
	File "/home/zzh/tensorflow/third_party/gpus/cuda_configure.bzl", line 976, in _create_local_cuda_repository
		_host_compiler_includes(repository_ctx, cc)
	File "/home/zzh/tensorflow/third_party/gpus/cuda_configure.bzl", line 145, in _host_compiler_includes
		get_cxx_inc_directories(repository_ctx, cc)
	File "/home/zzh/tensorflow/third_party/gpus/cuda_configure.bzl", line 120, in get_cxx_inc_directories
		set(includes_cpp)
depsets cannot contain mutable items
INFO: Elapsed time: 5.610s
FAILED: Build did NOT complete successfully (0 packages loaded)
    currently loading: tensorflow/tools/pip_package

```

**Comment by lukashann** [comment]
Have a look at this
https://github.com/tensorflow/tensorflow/issues/11859

using an older version of bazel (0.5.2) should fix the problem

**Comment by uglybachelor** [comment]
Solved the problem by rolling back to  bazel (0.5.2) .

**Comment by amoux** [comment]
I am glad I found this because i spent too much time trying to fix these related issues when building with bazel and tensorflow(GPU). Rolling back to Bazel (0.5.2) Worked for me!
